propagation/propagation.go

```diff
@@ -1,13 +1,35 @@
 package propagation
 
 import (
+	"context"
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
 	"net/url"
 	"strings"
+
+	"github.com/honeycombio/beeline-go/trace"
+	otelprop "go.opentelemetry.io/otel/api/propagation"
+)
+
+const (
+	HoneycombTracePropagationHTTPHeader = "X-Honeycomb-Trace"
+	HoneycombTracePropagationVersion    = 1
+	AmazonTracePropagationHTTPHeader    = "X-Amzn-Trace-Id"
 )
 
+type PropagationError struct {
```

Comment:
Why export this type?

Reply:
Lowercased. There are some other types and functions that might be exported and shouldn't. Will do a full audit before promoting this PR to ready to merge.